elastic/kibana
# [Cloud] Update support and user profile header menus

This PR updates the support and user profile dropdown **for cloud only**.

### How to test

* To simulate that we are running on cloud add the following in the `kibana.yml`

```
xpack.cloud.id: "foo"
xpack.cloud.base_url: "https://cloud.elastic.co"
xpack.cloud.organization_url: "/account/"
xpack.cloud.billing_url: "/billing/"
xpack.cloud.profile_url: "/user/settings/"
```

* Open the `maybe_add_cloud_links.ts` file and change L28

```diff
// x-pack/plugins/cloud_integrations/cloud_links/public/maybe_add_cloud_links/maybe_add_cloud_links.ts

     defer(() => security.authc.getCurrentUser())
       .pipe(
         // Check if user is a cloud user.
-        map((user) => user.elastic_cloud_user),
+        map((user) => true),
         // If user is not defined due to an unexpected error, then fail *open*.
         catchError(() => of(true)),
         filter((isElasticCloudUser) => isElasticCloudUser === true),
```

* Start Kibana
* Check that the dropdown menu items have been updated as suggested in the issue and that the links are correct

### Release notes for Cloud

There is a new Kibana configuration that needs to be set for the `Billing` link to appear in the menu

```
xpack.cloud.billing_url="/billing/"
```

### Screenshots


#### Support menu

**Cloud**

<img width="311" alt="Screenshot 2023-06-26 at 14 13 07" src="https://github.com/elastic/kibana/assets/2854616/47677587-cd15-4d63-a2a5-f7cf3f715523">

**Not cloud**
<img width="356" alt="Screenshot 2023-06-26 at 14 12 22" src="https://github.com/elastic/kibana/assets/2854616/b71d5ac0-4a1f-4cef-880a-61762b98c0e3">

#### User profile


**Cloud**

<img width="298" alt="Screenshot 2023-06-26 at 15 39 43" src="https://github.com/elastic/kibana/assets/2854616/dc6ccae9-28be-4b96-a029-3bd015aba9b1">

**Not cloud**
<img width="298" alt="Screenshot 2023-06-26 at 14 12 31" src="https://github.com/elastic/kibana/assets/2854616/b2ae5b4f-d136-45b0-83a7-14d10c0e2e68">


### TODO

In the linked issue we want to add a switch to change the Kibana theme darkmode from inside the user profile dropdown menu. I decided to split the work into [a separate issue](https://github.com/elastic/kibana/issues/160534) and keep this PR focused on updating the links + title for cloud.


Fixes https://github.com/elastic/kibana/issues/159419
